### PR TITLE
Ensure unit status get/set uses new leadership

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

The unit agent facade backend was not checking for leadership properly in the status get/set calls.
Update to use the new infrastructure.

## QA steps

Run a k8s model with a bad pod and check the app status is set.

### CODE REVIEW

File: apiserver/facades/agent/uniter/status.go [hunk 1]
```diff
@@ -22,13 +23,13 @@ type StatusAPI struct {
...
 // NewStatusAPI creates a new server-side Status setter API facade.
-func NewStatusAPI(st *state.State, getCanModify common.GetAuthFunc) *StatusAPI {
+func NewStatusAPI(st *state.State, getCanModify common.GetAuthFunc, leadershipChecker leadership.Checker) *StatusAPI {
 	// TODO(fwereade): so *all* of these have exactly the same auth
 	// characteristics? I think not.
 	unitSetter := common.NewStatusSetter(st, getCanModify)
 	unitGetter := common.NewStatusGetter(st, getCanModify)
-	serviceSetter := common.NewServiceStatusSetter(st, getCanModify)
-	serviceGetter := common.NewApplicationStatusGetter(st, getCanModify)
+	serviceSetter := common.NewApplicationStatusSetter(st, getCanModify, leadershipChecker)
```
Comment (anastasiamac) [inline apiserver/facades/agent/uniter/status.go]: I am wondering why there are 2?

Comment (wallyworld) [reply apiserver/facades/agent/uniter/status.go]: there's getter/setter for units and applications
the diff is a bit confusing

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**